spring/boot2-autoconfigure/src/main/java/com/linecorp/armeria/spring/AbstractArmeriaAutoConfiguration.java

```diff
@@ -107,6 +108,7 @@ public Server armeriaServer(
      * Wrap {@link Server} with {@link SmartLifecycle}.
      */
     @Bean
+    @ConditionalOnMissingBean(ArmeriaServerSmartLifecycle.class)
```

Comment:
The previous PR #4395 used `SmartLifecycle.class` for `@ConditionalOnMissingBean`.
It didn't work after Spring Boot is upgraded because the new Spring Boot registered a `SmartLifecycle` as a bean.
https://github.com/spring-projects/spring-boot/blob/main/spring-boot-project/spring-boot/src/main/java/org/springframework/boot/context/logging/LoggingApplicationListener.java#L460
So `ArmeriaServerSmartLifecycle` is added to instanciate `ArmeriaServerGracefulShutdownLifecycle` for the condition where `RetryableArmeriaServerGracefulShutdownLifecycle` is missing.